### PR TITLE
Improve RabbitMQ producer example

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose
This fix is to address the following concerns.
- The existing example is too complex
- There is no example with the use of exchanges 
- The existing producer does not match the consumer example (the current producer is sending messages to MyQueue1 and MyQueue2, the consumer is listening to MyQueue)

Running the new example with the [consumer example](https://github.com/ballerina-platform/ballerina-distribution/blob/master/examples/rabbitmq-consumer/rabbitmq_consumer.bal) will work seamlessly
Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/478

### CODE REVIEW

File: examples/rabbitmq-producer/rabbitmq_producer.bal [hunk 1]
```diff
@@ -5,53 +5,33 @@ public function main() {
     // Creates a ballerina RabbitMQ connection that allows re-usability if necessary.
```
Comment (praneesha) [inline examples/rabbitmq-producer/rabbitmq_producer.bal:5]:
```suggestion
    // Creates a Ballerina RabbitMQ connection that allows re-usability if necessary.
```

File: examples/rabbitmq-producer/rabbitmq_producer.bal [hunk 2]
```diff
@@ -5,53 +5,33 @@ public function main() {
...
     rabbitmq:Connection connection = new ({host: "localhost", port: 5672});
 
-    // Creates multiple ballerina RabbitMQ channels.
-    rabbitmq:Channel newChannel1 = new (connection);
-    rabbitmq:Channel newChannel2 = new (connection);
+    // Creates a ballerina RabbitMQ channel.
```
Comment (praneesha) [inline examples/rabbitmq-producer/rabbitmq_producer.bal:8]:
```suggestion
    // Creates a Ballerina RabbitMQ channel.
```

File: examples/rabbitmq-producer/rabbitmq_producer.bal [hunk 3]
```diff
@@ -5,53 +5,33 @@ public function main() {
...
+    rabbitmq:Channel newChannel = new (connection);
 
-    // Declares the queue, MyQueue1.
-    var queueResult1 = newChannel1->queueDeclare({queueName: "MyQueue1"});
-    if (queueResult1 is error) {
-        io:println("An error occurred while creating the MyQueue1 queue.");
+    // Declares the queue, MyQueue.
```
Comment (praneesha) [inline examples/rabbitmq-producer/rabbitmq_producer.bal:11]:
```suggestion
    // Declares a queue named `MyQueue`.
```

File: examples/rabbitmq-producer/rabbitmq_producer.bal [hunk 4]
```diff
@@ -5,53 +5,33 @@ public function main() {
...
+            newChannel->queueDeclare({queueName: "MyQueue"});
+    if (queueResult is error) {
+        io:println("An error occurred while creating the queue.");
     }
 
-    // Declares the queue, MyQueue2.
-    var queueResult2 = newChannel2->queueDeclare({queueName: "MyQueue2"});
-    if (queueResult2 is error) {
-        io:println("An error occurred while creating the MyQueue2 queue.");
+    // Declares the direct exchange, MyExchange.
```
Comment (praneesha) [inline examples/rabbitmq-producer/rabbitmq_producer.bal:18]:
```suggestion
    // Declares a direct exchange named `MyExchange`.
```

File: examples/rabbitmq-producer/rabbitmq_producer.bal [hunk 5]
```diff
@@ -5,53 +5,33 @@ public function main() {
...
-    // Publishes the message using newChannel2 and the same routing key named MyQueue1.
-    worker w2 {
-        var sendResult = newChannel2->basicPublish("Hello from Ballerina",
-                                        "MyQueue1");
-        if (sendResult is error) {
-            io:println("An error occurred while sending the message to " +
-                    "MyQueue1 using newChannel2.");
-        }
-    }
+    // Binds MyQueue to MyExchange with the routing key named example-key.
```
Comment (praneesha) [inline examples/rabbitmq-producer/rabbitmq_producer.bal:26]:
```suggestion
    // Binds `MyQueue` to `MyExchange` with the routing key named `example-key`.
```

File: examples/rabbitmq-producer/rabbitmq_producer.bal [hunk 6]
```diff
@@ -5,53 +5,33 @@ public function main() {
...
-    // Publishing messages to different routing keys using the same channel.
-    // Publishes the message using newChannel1 to a different routing key named MyQueue2.
-    worker w3 {
-        var sendResult = newChannel1->basicPublish("Hello from Ballerina",
-                                        "MyQueue2");
-        if (sendResult is error) {
-            io:println("An error occurred while sending the message to " +
-                    "MyQueue2 using newChannel1.");
-        }
+    // Publishes the message to the routing key example-key using newChannel.
```
Comment (praneesha) [inline examples/rabbitmq-producer/rabbitmq_producer.bal:30]:
```suggestion
    // Publishes the message to the `example-key` using the `newChannel`.
```

**Review by praneesha — APPROVED** [review]

**Comment by aashikam** [comment]
Closing since the API change, this is invalid.
Ref: https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/59

**Pull request closed without merging**